tests/test_vagrant.py

```diff
@@ -603,6 +608,19 @@ def test_provisioning(vm_dir):
     assert test_file_contents == "foo", "The test file should contain 'foo'"
 
 
+@pytest.mark.parametrize("vm_dir", (SHELL_PROVISION_FAIL_VAGRANTFILE,), indirect=True)
+def test_provisioning_no_destory_on_error(vm_dir):
+    """
+    Test support for 'vagrant up --no-destroy-on-error'.
+    """
+    v = vagrant.Vagrant(vm_dir)
+    v.up(destroy_on_error=True)
```

Comment:
This alone will lead to test failures, since the call to vagrant up is done with ``subprocess.check_call``. We know it'll fail at provisionning stage due to the ``exit 1`` call so ``v.up`` will raise an exception. You should use something like that:
```
    try:
        v.up(destroy_on_error=True)
    except subprocess.CalledProcessError:
        pass
```
for both ``v.up`` calls.